mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/SocketPolicy.java

```diff
@@ -28,6 +28,11 @@
   /**
    * Close the socket after the response. This is the default HTTP/1.0
    * behavior.
+   *
+   * <p>Note: Be careful if this used in a test and it is not the final queued request. The client
```

Comment:
The tests appear to be testing "what happens if a pooled connection has gone bad?", not "what happens if the server closes the connection at an arbitrary point during the request / response". 

You agree it's non-deterministic? Just to be clear in case my comments didn't explain:

I've seen this policy used in two ways:

1) Advertise a body as being N bytes, but disconnect before delivering the full N bytes.
2) Advertise a body as being N bytes, deliver the full N bytes, then (after time T) disconnect.

(1) is fine, since the client is blocked waiting for N bytes.
(2) is problematic if the test is trying to test "what happens if a pooled connection is no longer valid" : the client receives the N bytes, then the result depends on how long T is. Usually the test is expecting the connection to be closed by the time a second request is issued. However, because time T is varied, the test can move on to make another request and have the connection closed in the middle or after the client has issued the request.

If the server closes the connection before request 2 has been made the mock server needs to be set up to expect 2 requests. If it closes it after request 2 has been made and OkHttp retries, then it needs to be set up to expect 3 requests.